### PR TITLE
task(admin-sever): Move token audit scripts out of auth-server

### PR DESCRIPTION
**Pull request by dschom**
## Because:
- We want to run audits from the admin server

## This Pull Request:
- Moves the script from the auth server to admin server
- Moves associated tests from the auth server to the admin server
- Stops stripe service listeners so as to close some open handles
- Enable typescript compilation on scripts folder and fixes a couple pre-existing errors

## Issue that this pull request solves

Closes: FXA-6548

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

## Other information (Optional)

This was requested by SRE. The admin server is a better fit for running this kind of script, because the admin server runs on a single instance.

### CODE REVIEW

File: packages/fxa-admin-server/scripts/audit-tokens.spec.ts [hunk 1]
```diff
@@ -4,20 +4,40 @@
...
-} = require('../../scripts/audit-tokens');
-const mocks = require(`../../test/mocks`);
-const config = require('../../config').getProperties();
-const log = mocks.mockLog();
-const { Account } = require('fxa-shared/db/models/auth/account');
-const { clearDb, scaffoldDb, connectToDb } = require('./db-helpers');
-
+import { assert } from 'chai';
+import util from 'node:util';
+import path from 'path';
```
Comment (pdehaan) [inline packages/fxa-admin-server/scripts/audit-tokens.spec.ts:9]: micro nit: love the `node:` prefix on the "node:util" import on L8, why no "node:path" here?
but the tiniest of nits since I don't think we consistently use `node:` for internal packages (but I think I did find a way to enforce that a while ago while poking at some ESLint importer rules and forbidden modules config).

File: packages/fxa-admin-server/scripts/db-helpers.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+} from 'fxa-shared/db/models/auth';
+import crypto from 'crypto';
+import { setupAuthDatabase } from 'fxa-shared/db';
+import { ILogger } from 'fxa-shared/log';
+
+const toRandomBuff = (size: number) =>
+  uuidTransformer.to(crypto.randomBytes(size).toString('hex'));
+
+export const toZeroBuff = (size: number) =>
+  Buffer.from(Array(size).fill(0)).toString('hex');
```
Comment (pdehaan) [inline packages/fxa-admin-server/scripts/db-helpers.ts:21]: I might be confused by these two functions (`toRandomBuff()` and `toZeroBuff()`).
If I understand correctly `toRandomBuff()` will return a Buffer, but `toZeroBuff()` might be returning a string?
If so, is the function name misleading, or should we just return the raw buffer and then call `.toString('hex')` separately (like on L28-29 below?

Not sure if these need explicit return types either. I'm sure TypeScript is great at inferring types, but explicit ":string" or ":buffer" might be useful if they are indeed different return types.

File: packages/fxa-admin-server/scripts/audit-tokens.spec.ts (hunk 1 above)
Comment (dschom) [reply packages/fxa-admin-server/scripts/audit-tokens.spec.ts:9]: I wish I could take credit for the `node:util`, but I think vscode automatically generated these imports. Good call out on consistency issue though.

File: packages/fxa-admin-server/scripts/audit-tokens.spec.ts [hunk 3]
```diff
@@ -4,40 +4,61 @@
...
+  'info',
+  'notifyAttachedServices',
+  'warn',
+  'summary',
+  'trace',
+  'debug',
+];
+function mockObject(methodNames: string[], baseObj?: any) {
+  return (methods?: any) => {
+    methods = methods || {};
```
Comment (LZoog) [inline packages/fxa-admin-server/scripts/audit-tokens.spec.ts:32]:
```
return (methods?: any) => {
    methods = methods || {};
```

Can this be `return (methods = {}) => {`?

File: packages/fxa-admin-server/scripts/db-helpers.ts [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+
+  await Account.create(act);
+
+  await SessionToken.create(token);
+  await Device.create(device(uid, token.id, createdAt));
+  const code = signInCode(uid, createdAt);
+  await SignInCodes.knexQuery().insert(code);
+}
+
+// export async function connectToDb(config, log) {
```
Comment (LZoog) [inline packages/fxa-admin-server/scripts/db-helpers.ts:105]: Was the need for this replaced with simply `bindKnex(config.database.fxa);`? 😮 (Can we remove these commented out lines?)

File: packages/fxa-shared/package.json [hunk 5]
```diff
@@ -54,9 +54,10 @@
   },
   "scripts": {
     "postinstall": "yarn build || true",
-    "build": "tsc --build && yarn copy-assets",
-    "compile": "tsc --noEmit",
+    "build": "tsc --build && yarn copy-assets && yarn copy-sql",
+    "compile": "tsc --noEmit && yarn copy-assets && yarn copy-sql",
     "copy-assets": "cp -r ./db/luaScripts ./dist/db/luaScripts",
+    "copy-sql": "find test -name \\*.sql -exec cp \\{\\} dist/\\{\\} \\;",
```
Comment (LZoog) [inline packages/fxa-shared/package.json:60]: This is probably because I haven't reviewed any other token audit PRs but why this change in `fxa-shared`?

**Review by LZoog — APPROVED** [review]
Couple questions but LGTM!

File: packages/fxa-admin-server/scripts/db-helpers.ts (hunk 4 above)
Comment (dschom) [reply packages/fxa-admin-server/scripts/db-helpers.ts:105]: Yes, good catch.

File: packages/fxa-shared/package.json (hunk 5 above)
Comment (dschom) [reply packages/fxa-shared/package.json:60]: I was amazed this hadn't caused a problem previously, but without this the the dist folder wouldn't contain `.sql` files, and some of the code expects those files to be there. This [line](https://github.com/mozilla/fxa/blob/c3de9f95e2cd92694039782415248c23d1f66874/packages/fxa-shared/test/db/models/profile/helpers.ts#L15) would consistently fail, without this addition.

The issue crept into the admin server, because we now compile scripts folder. We need to do this, so that SRE can run the script with node sans any ts compilation. The `admin-server/scripts/email-bounct.ts` [file](https://github.com/mozilla/fxa/blob/c3de9f95e2cd92694039782415248c23d1f66874/packages/fxa-admin-server/scripts/email-bounce.ts#L13) references fxa-shared/test/db/models/profile/helpers.ts, which is how that got tangled up and ultimately resulted in a build error.

Anyways, I think the `.sql` files should end up in the dist folder if the compiled code anticipates they are there.

File: packages/fxa-admin-server/src/subscriptions/stripe.service.ts [hunk 6]
```diff
@@ -56,6 +56,10 @@ export class StripePaymentConfigManagerService extends PaymentConfigManager {
 
     super(config, firestore, logger);
   }
+
+  async onModuleDestroy() {
+    await this.stopListeners();
```
Comment (dschom) [inline packages/fxa-admin-server/src/subscriptions/stripe.service.ts:61]: This was important to prevent an error from happening during teardown. The error would complain about a bulkwrite operation not being closed.
